CDL-Dryad/dryad-app
# Use S3 for Merritt/SWORD submission

For https://github.com/CDL-Dryad/dryad-product-roadmap/issues/1078

- Removes the ZIP files that were previously sent to Merritt
- Puts all Merritt files *except* the main `manifest.checkm` into S3, and sends them to Merritt
- Adds a utility class for working with S3

Known problem: This does not correctly handle some special characters, including the 4-byte Unicode characters.

To test:
- create a dataset, where some files are stored in S3 (Just upload the files through the UI, since it's all merged in)
- when you finish the submission, you should see the `/manifest` directory being created in S3, under the resource's main directory
- Merritt should correctly process the files
- when the notifier runs, it should delete all files in `/manifest` and `/data`, but the other directories will remain (since they need to be cleaned up by the Zenodo processor).
